src/components/subscriptions/data/hooks.js

```diff
@@ -0,0 +1,159 @@
+import { useEffect, useMemo, useState } from 'react';
+
+import LicenseManagerApiService from './service';
+import NewRelicService from '../../../data/services/NewRelicService';
+import {
+  licenseStatusByTab,
+  NETWORK_ERROR_MESSAGE,
+  SUBSCRIPTION_USERS,
+  SUBSCRIPTION_USERS_OVERVIEW,
+  SUBSCRIPTIONS,
+} from './constants';
+import { camelCaseObject } from '../../../utils';
+
+/*
+This hook provides all subscription data for the authenticated user and given enterprise customer UUID.
+ */
+export const useSubscriptions = ({ enterpriseId, errors, setErrors }) => {
+  const [subscriptions, setSubscriptions] = useState({
+    results: [],
+    count: 0,
+    next: null,
+    previous: null,
+  });
+
+  const forceRefresh = () => {
+    setSubscriptions({ ...subscriptions });
+  };
+
+  useMemo((page = 1) => {
+    LicenseManagerApiService.fetchSubscriptions({ enterprise_customer_uuid: enterpriseId, page })
+      .then((response) => {
+        const { data: subscriptionsData } = response;
+        const subscriptionsList = subscriptionsData?.results || [];
+        subscriptionsData.results = subscriptionsList.filter((
+          subscription => subscription.is_active
+        ));
+        setSubscriptions(camelCaseObject(subscriptionsData));
+      })
+      .catch((err) => {
+        NewRelicService.logAPIErrorResponse(err);
```

Comment:
in general, I thought a good improvement here (again not for this PR!) will be to call a more abstract method like logErrorResponse() or something like it and have it handle NewRelic-ness but we can have that be considered separately

Reply:
thanks filed https://openedx.atlassian.net/browse/ENT-3852